### PR TITLE
Overview changes

### PR DESCRIPTION
**Pull request by Nkozlare**
Add 5 actual tests. 2 of which are for testing the selector. 
When a product id is changed, it changes what item is displayed on the product view.
Implements use of createContext and useContext as well as compound rendering (in the video).

### CODE REVIEW

File: client/src/components/overview/GalleryView.jsx [hunk 1]
```diff
@@ -39,6 +39,33 @@ const StyledOverviewGrid = styled.div`
...
+  border-radius: 50%;
+  background-size: cover;
+  margin: 5px;
+  cursor: pointer;
+  border: 2px solid gray;
+  transition: 0.3s;
+  &:hover {
+    border: 2px solid white;
+  }
+`;
```
Comment (Nkozlare) [inline client/src/components/overview/GalleryView.jsx:69]: This code was just moved from another file. No need to look over it.

File: client/src/components/overview/GalleryView.jsx [hunk 2]
```diff
@@ -232,13 +260,28 @@ function Overview({ currentStyleId, setCurrentStyleId }) {
...
+                    setImgIndex(index);
+                  }}
+                  value={index} key={index} style={{backgroundImage: `url(${photo.url})`}}></StyledIcon>
+                })
+              }
+            </StyledIconSection>
+          </Expanded>
+        </ExpandedContext.Provider>
+        {/* <Expanded photos={styles[styleIndex].photos} expanded={expanded} setView={setView} imgIndex = {imgIndex} setImgIndex={setImgIndex}/> */}
       </StyledOverviewGrid>
```
Comment (Nkozlare) [inline client/src/components/overview/GalleryView.jsx:281]: This is the compound rendering area and won't make much sense until you watch the video. It works though.

File: client/src/components/overview/overview-components/Carousel.jsx [hunk 3]
```diff
@@ -196,11 +196,11 @@ export default function OverviewCarousel({ photos, expanded, setView, imgIndex,
...
+      {y >= ((photos.length * 135) - 800) ? <DownButton data-testid='down-arrow' onClick={goDown}><FontAwesomeIcon icon={faChevronDown} /></DownButton> : (<div></div>)}
     </ThumbnailSelector>
     {/* these two ternary statements are to render in either a button or nothing depending on what position the gallery is in */}
-    {x === 0 ? (<div></div>) : (<LeftButton onClick={goLeft}><FontAwesomeIcon icon={faChevronLeft} /></LeftButton>)}
-    {x > (-100 * photos.length + 100) ? (<RightButton onClick={goRight}><FontAwesomeIcon icon={faChevronRight} /></RightButton>) : (<div></div>)}
+    {x === 0 ? (<div></div>) : (<LeftButton data-testid='right-arrow' onClick={goLeft}><FontAwesomeIcon icon={faChevronLeft} /></LeftButton>)}
+    {x > (-100 * photos.length + 100) ? (<RightButton data-testid='left-arrow' onClick={goRight}><FontAwesomeIcon icon={faChevronRight} /></RightButton>) : (<div></div>)}
   </StyledOverviewCarousel>
   )
 }
```
Comment (Nkozlare) [inline client/src/components/overview/overview-components/Carousel.jsx:206]: This was for writing tests, adding the test ids

File: client/src/components/overview/overview-components/Stars.jsx [hunk 4]
```diff
@@ -19,7 +27,9 @@ export default function OverviewStars({ stars, reviewCount }) {
           starHoverColor="yellow"
           name='rating'
         />
-        <div style={{cursor: 'pointer', textDecoration: 'none', color: '#6699cc'}}>Read all {reviewCount} reviews</div>
+        <div onClick={() => {
+          handleScroll("reviews")
+        }}style={{cursor: 'pointer', textDecoration: 'none', color: '#6699cc'}}>Read all {reviewCount} reviews</div>
       </StyledOverviewStars>
   );
 }
```
Comment (Nkozlare) [inline client/src/components/overview/overview-components/Stars.jsx:35]: This was where I added the ability to clink on the reviews count in the overview that would scroll down to the reviews

**Review by Chrisgood2go — APPROVED** [review]
no comment except awesome job!

**Pull request merged**